### ISSUE TITLE
The model dose not converge

### ISSUE THREAD

**Issue by marvin521** [chenzhaomin123/ML_GCN#4]
I have run the training code. However, I found that it is hard to converge during the training process. The following are hyperparameters I used:

python3 demo_voc2007_gcn.py data/voc --image-size 448 --batch-size 16
pytorch 1.0
learning rate: 0.01

are there any parameters I should modify to re-implement the results shown in the paper?

**Comment by marvin521** [comment]
The above problem solved by setting values to learning rate

**Comment by Abhinav43** [comment]
Can you provide instructions how to run code for training and how to generate adj matrix?

**Comment by marvin521** [comment]
@Abhinav43 . run the following scripts to start training:
  python3 demo_voc2007_gcn.py data/voc --image-size 448 --batch-size 32

I did not generate adj matrix, just used the matrix provided by the authors

**Comment by Abhinav43** [comment]
@marvin521  If I want to use my own data how can I generate adj matrix?

**Comment by monk1337** [comment]
@Abhinav43 
You have to count occurrence of labels in dataset and then create conditional probability between labels. 

For more details, check the paper.

**Comment by stevehjc** [comment]
> @Abhinav43 . run the following scripts to start training:
> python3 demo_voc2007_gcn.py data/voc --image-size 448 --batch-size 32
> 
> I did not generate adj matrix, just used the matrix provided by the authors

The mAP does not improve when I training.
I set `lr=0.05`,`batch_size=16`,use 2 GPU for voc2007 training, but the mAP is always around 8.827.(epoch1~15)
```shell
[dataset] read data/voc/files/VOC2007/classification_trainval.csv
[dataset] VOC 2007 classification set=trainval number of classes=20  number of images=5011
[dataset] read data/voc/files/VOC2007/classification_test.csv
[dataset] VOC 2007 classification set=test number of classes=20  number of images=4952
<torchvision.transforms.transforms.Compose object at 0x7f64390be0f0>
lr:0.00500
Training: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 314/314 [02:04<00:00,  2.01s/it]
Epoch: [0]	Loss 2.2040	mAP 8.095
OP: 0.1741	OR: 0.0226	OF1: 0.0400	CP: 0.0782	CR: 0.0108	CF1: 0.0190
Test: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 310/310 [00:57<00:00,  1.60it/s]
Test: 	 Loss 2.1294	 mAP 9.103
OP: 0.0000	OR: 0.0000	OF1: 0.0000	CP: 0.0000	CR: 0.0000	CF1: 0.0000
OP_3: 0.0000	OR_3: 0.0000	OF1_3: 0.0000	CP_3: 0.0000	CR_3: 0.0000	CF1_3: 0.0000
save model checkpoint/voc2007_exp3_lr0005_bs16_gpu01/checkpoint.pth.tar
 *** best=9.103
# ...
# ...
# ...
lr:0.00500
Training: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 314/314 [05:55<00:00,  1.15it/s]
Epoch: [15]	Loss 2.1820	mAP 8.827
OP: 0.0000	OR: 0.0000	OF1: 0.0000	CP: 0.0000	CR: 0.0000	CF1: 0.0000
Test: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 310/310 [01:33<00:00,  4.24it/s]
Test: 	 Loss 2.1294	 mAP 9.070
OP: 0.0000	OR: 0.0000	OF1: 0.0000	CP: 0.0000	CR: 0.0000	CF1: 0.0000
OP_3: 0.0000	OR_3: 0.0000	OF1_3: 0.0000	CP_3: 0.0000	CR_3: 0.0000	CF1_3: 0.0000
save model checkpoint/voc2007_exp3_lr0005_bs16_gpu01/checkpoint.pth.tar
 *** best=9.154
```

**Comment by Alex661** [comment]
I met the same problem, did you slove it?
